OpenRA.Mods.Common/Activities/Air/Land.cs

```diff
@@ -21,18 +21,33 @@ public class Land : Activity
 		readonly Aircraft aircraft;
 		readonly bool requireSpace;
 		readonly Actor ignoreActor;
+		readonly WDist landAltitude;
+		readonly bool ignoreTarget;
 
 		bool landingInitiated;
 		bool soundPlayed;
 
-		public Land(Actor self, Target t, bool requireSpace, Actor ignoreActor = null)
+		public Land(Actor self, Target t, bool requireSpace, WDist landAltitude, Actor ignoreActor = null)
 		{
 			target = t;
 			aircraft = self.Trait<Aircraft>();
 			this.requireSpace = requireSpace;
 			this.ignoreActor = ignoreActor;
+			this.landAltitude = landAltitude != WDist.Zero ? landAltitude : aircraft.Info.LandAltitude;
 		}
 
+		public Land(Actor self, Target t, bool requireSpace, Actor ignoreActor = null)
+			: this(self, t, requireSpace, WDist.Zero, ignoreActor) { }
+
+		public Land(Actor self, bool requireSpace, WDist landAltitude, Actor ignoreActor = null)
+			: this(self, Target.FromPos(self.CenterPosition), requireSpace, landAltitude, ignoreActor)
+		{
+			ignoreTarget = true;
+		}
+
+		public Land(Actor self, bool requireSpace, Actor ignoreActor = null)
+			: this(self, requireSpace, WDist.Zero, ignoreActor) { }
```

Comment:
Four overloads seems a bit unwieldy. Maybe it is possible to use named optional arguments for `target` and `landAltitude` instead of overloads?

Reply:
`Target` and `WDist` don't really support that, but I refactored `landAltitude` to not need a parameter and removed `requireSpace` as you suggested above, which was enough to get down to two overloads.